### ISSUE TITLE
Train Loss is NAN after 13 epoch

### ISSUE THREAD

**Issue by heartQi** [wenet-e2e/wesep#18]
After modify some config, train loss become NAN after 13 epoch
num_workers = 1
batch_size = 4

[ INFO : 2025-01-13 01:09:33,989 ] - Epoch 13 Train info train_loss -6.314864508356337
[ INFO : 2025-01-13 01:09:33,989 ] - Epoch 13 Val info val_loss -6.05645558488369
[ INFO : 2025-01-13 01:12:25,280 ] - |     TRAIN|        14|       100|   -6.2999|0.00072586|
[ INFO : 2025-01-13 01:13:48,348 ] - |     TRAIN|        14|       200|   -6.4173|0.00072534|
[ INFO : 2025-01-13 01:15:15,533 ] - |     TRAIN|        14|       300|   -6.4377|0.00072483|
[ INFO : 2025-01-13 01:16:38,424 ] - |     TRAIN|        14|       400|   -6.3899|0.00072432|
[ INFO : 2025-01-13 01:18:04,831 ] - |     TRAIN|        14|       500|   -6.4196| 0.0007238|
[ INFO : 2025-01-13 01:19:27,955 ] - |     TRAIN|        14|       600|    -6.485|0.00072329|
[ INFO : 2025-01-13 01:22:18,272 ] - |     TRAIN|        14|       700|   -6.4901|0.00072278|
[ INFO : 2025-01-13 01:23:40,060 ] - |     TRAIN|        14|       800|   -6.4564|0.00072227|
[ INFO : 2025-01-13 01:25:01,244 ] - |     TRAIN|        14|       900|   -6.4357|0.00072176|
[ INFO : 2025-01-13 01:26:28,100 ] - |     TRAIN|        14|      1000|   -6.4473|0.00072125|
[ INFO : 2025-01-13 01:27:56,007 ] - |     TRAIN|        14|      1100|   -6.4738|0.00072074|
[ INFO : 2025-01-13 01:29:19,003 ] - |     TRAIN|        14|      1200|   -6.5021|0.00072023|
[ INFO : 2025-01-13 01:32:13,902 ] - |     TRAIN|        14|      1300|    -6.532|0.00071972|
[ INFO : 2025-01-13 01:33:40,630 ] - |     TRAIN|        14|      1400|   -6.5525|0.00071921|
[ INFO : 2025-01-13 01:35:03,001 ] - |     TRAIN|        14|      1500|   -6.5299| 0.0007187|
[ INFO : 2025-01-13 01:36:27,902 ] - |     TRAIN|        14|      1600|   -6.5483|0.00071819|
[ INFO : 2025-01-13 01:37:48,451 ] - |     TRAIN|        14|      1700|   -6.5733|0.00071768|
[ INFO : 2025-01-13 01:39:20,407 ] - |     TRAIN|        14|      1800|   -6.5702|0.00071718|
[ INFO : 2025-01-13 01:42:04,345 ] - |     TRAIN|        14|      1900|   -6.5762|0.00071667|
[ INFO : 2025-01-13 01:43:32,368 ] - |     TRAIN|        14|      2000|   -6.5762|0.00071616|
[ INFO : 2025-01-13 01:44:52,891 ] - |     TRAIN|        14|      2100|   -6.5889|0.00071565|
[ INFO : 2025-01-13 01:46:22,952 ] - |     TRAIN|        14|      2200|   -6.5855|0.00071515|
[ INFO : 2025-01-13 01:47:47,055 ] - |     TRAIN|        14|      2300|   -6.5959|0.00071464|
[ INFO : 2025-01-13 01:49:25,271 ] - |     TRAIN|        14|      2400|   -6.6181|0.00071414|
[ INFO : 2025-01-13 01:50:49,177 ] - |     TRAIN|        14|      2500|   -6.6321|0.00071363|
[ INFO : 2025-01-13 01:53:32,470 ] - |     TRAIN|        14|      2600|   -6.6306|0.00071313|
[ INFO : 2025-01-13 01:54:53,155 ] - |     TRAIN|        14|      2700|   -6.6279|0.00071262|
[ INFO : 2025-01-13 01:56:17,618 ] - |     TRAIN|        14|      2800|   -6.6251|0.00071212|
[ INFO : 2025-01-13 01:57:39,873 ] - |     TRAIN|        14|      2900|       nan|0.00071161|
[ INFO : 2025-01-13 01:59:05,280 ] - |     TRAIN|        14|      3000|       nan|0.00071111|
[ INFO : 2025-01-13 02:00:29,425 ] - |     TRAIN|        14|      3100|       nan|0.00071061|
[ INFO : 2025-01-13 02:03:17,187 ] - |     TRAIN|        14|      3200|       nan|0.00071011|
[ INFO : 2025-01-13 02:04:36,906 ] - |     TRAIN|        14|      3300|       nan| 0.0007096|
[ INFO : 2025-01-13 02:06:02,308 ] - |     TRAIN|        14|      3400|       nan| 0.0007091|

**Comment by sangeet2020** [comment]
hi @mrjunjieli @heartQi ,

I am experiencing same problem. No change in setting was made except for the batch size which I used 1. Also, I am using 2 GPUs. I encounter Nan loss right around 3rd epoch itself. 
What could be possible way to fix this?

**Comment by mrjunjieli** [comment]
i will check.  does this problem occur when you  set batch_size to more than 1?

**Comment by sangeet2020** [comment]
My hardware availability doesnt allow that yet. I encountered memory error using 4 GPUS/8 GPUS with Batch size 1. Any setting with a batch size more than 1 also leads to memory error.

**Comment by sriyachakravarthy** [comment]
Hi, I am facing the same issue. In the paper, it is mentioned that the training was done on Libri2mix-360 and under 8kHz sampling rate. Can we use this in the recipe instead?

**Comment by sangeet2020** [comment]
I had to increase my batch size to 4 and enable mixed precision which resolved the NaN loss.

@sriyachakravarthy adding to your comment- its relatively easier to train on 8 kHz comapred to 16 kHz since you effictively reduce the BW by half. So you might want to try further increasing the batch size to 8 or more depending upon your compute resources.

**Comment by sriyachakravarthy** [comment]
Yes, But when i changed sr=8khz, i was getting the error as linked https://github.com/wenet-e2e/wesep/issues/25#issue-2883517367

**Comment by ferugit** [comment]
Hi, 

The same problem here. I changed several things like bach_size, speaker model (SimAMResnet34), raw audio processing instead of shards. Here is my config:

```bash
[ INFO : 2025-12-31 15:10:08,113 ] - exp_dir is: exp/BSRNN/no_spk_transform-multiply_fuse
[ INFO : 2025-12-31 15:10:08,113 ] - <== Passed Arguments ==>
[ INFO : 2025-12-31 15:10:08,114 ] - {'clip_grad': 5.0,
[ INFO : 2025-12-31 15:10:08,114 ] -  'config': 'confs/bsrnn_simamresnet.yaml',
[ INFO : 2025-12-31 15:10:08,114 ] -  'data_type': 'raw',
[ INFO : 2025-12-31 15:10:08,114 ] -  'dataloader_args': {'batch_size': 2,
[ INFO : 2025-12-31 15:10:08,114 ] -                      'drop_last': True,
[ INFO : 2025-12-31 15:10:08,114 ] -                      'num_workers': 2,
[ INFO : 2025-12-31 15:10:08,114 ] -                      'pin_memory': True,
[ INFO : 2025-12-31 15:10:08,114 ] -                      'prefetch_factor': 6},
[ INFO : 2025-12-31 15:10:08,114 ] -  'dataset_args': {'SSA_enroll_prob': 0,
[ INFO : 2025-12-31 15:10:08,114 ] -                   'chunk_len': 48000,
[ INFO : 2025-12-31 15:10:08,114 ] -                   'fbank_args': {'dither': 1.0,
[ INFO : 2025-12-31 15:10:08,114 ] -                                  'frame_length': 25,
[ INFO : 2025-12-31 15:10:08,114 ] -                                  'frame_shift': 10,
[ INFO : 2025-12-31 15:10:08,114 ] -                                  'num_mel_bins': 80},
[ INFO : 2025-12-31 15:10:08,114 ] -                   'noise_enroll_prob': 0,
[ INFO : 2025-12-31 15:10:08,114 ] -                   'noise_lmdb_file': './data/musan/lmdb',
[ INFO : 2025-12-31 15:10:08,114 ] -                   'noise_prob': 0,
[ INFO : 2025-12-31 15:10:08,114 ] -                   'resample_rate': 16000,
[ INFO : 2025-12-31 15:10:08,115 ] -                   'reverb_enroll_prob': 0,
[ INFO : 2025-12-31 15:10:08,115 ] -                   'sample_num_per_epoch': 0,
[ INFO : 2025-12-31 15:10:08,115 ] -                   'shuffle': True,
[ INFO : 2025-12-31 15:10:08,115 ] -                   'shuffle_args': {'shuffle_size': 2500},
[ INFO : 2025-12-31 15:10:08,115 ] -                   'speaker_feat': True,
[ INFO : 2025-12-31 15:10:08,115 ] -                   'specaug_enroll_prob': 0},
[ INFO : 2025-12-31 15:10:08,115 ] -  'enable_amp': False,
[ INFO : 2025-12-31 15:10:08,115 ] -  'exp_dir': 'exp/BSRNN/no_spk_transform-multiply_fuse',
[ INFO : 2025-12-31 15:10:08,115 ] -  'gpus': [0],
[ INFO : 2025-12-31 15:10:08,115 ] -  'log_batch_interval': 100,
[ INFO : 2025-12-31 15:10:08,115 ] -  'loss': 'SISDR',
[ INFO : 2025-12-31 15:10:08,115 ] -  'loss_args': {},
[ INFO : 2025-12-31 15:10:08,115 ] -  'model': {'tse_model': 'BSRNN'},
[ INFO : 2025-12-31 15:10:08,115 ] -  'model_args': {'tse_model': {'feat_type': 'consistent',
[ INFO : 2025-12-31 15:10:08,115 ] -                               'feature_dim': 128,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'joint_training': True,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'multi_fuse': False,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'multi_task': False,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'num_repeat': 6,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spk_args': {'embed_dim': 256},
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spk_emb_dim': 256,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spk_feat': True,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spk_fuse_type': 'multiply',
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spk_model': 'SimAM_ResNet34_ASP',
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spk_model_freeze': False,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spk_model_init': './wespeaker_models/voxblink2_samresnet34_ft/avg_model.pt',
[ INFO : 2025-12-31 15:10:08,115 ] -                               'spksInTrain': 251,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'sr': 16000,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'stride': 128,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'use_spk_transform': False,
[ INFO : 2025-12-31 15:10:08,115 ] -                               'win': 512}},
[ INFO : 2025-12-31 15:10:08,115 ] -  'model_init': {'discriminator': None, 'spk_model': None, 'tse_model': None},
[ INFO : 2025-12-31 15:10:08,115 ] -  'num_avg': 10,
[ INFO : 2025-12-31 15:10:08,115 ] -  'num_epochs': 150,
[ INFO : 2025-12-31 15:10:08,115 ] -  'optimizer': {'tse_model': 'Adam'},
[ INFO : 2025-12-31 15:10:08,115 ] -  'optimizer_args': {'tse_model': {'lr': 0.001, 'weight_decay': 0.0001}},
[ INFO : 2025-12-31 15:10:08,115 ] -  'save_epoch_interval': 1,
[ INFO : 2025-12-31 15:10:08,115 ] -  'scheduler': {'tse_model': 'ExponentialDecrease'},
[ INFO : 2025-12-31 15:10:08,115 ] -  'scheduler_args': {'tse_model': {'final_lr': 2.5e-05,
[ INFO : 2025-12-31 15:10:08,115 ] -                                   'initial_lr': 0.001,
[ INFO : 2025-12-31 15:10:08,115 ] -                                   'warm_from_zero': False,
[ INFO : 2025-12-31 15:10:08,116 ] -                                   'warm_up_epoch': 0}},
[ INFO : 2025-12-31 15:10:08,116 ] -  'seed': 42,
[ INFO : 2025-12-31 15:10:08,116 ] -  'train_data': 'data/both/train-100/raw.list',
[ INFO : 2025-12-31 15:10:08,116 ] -  'train_spk2utt': 'data/both/train-100/spk2enroll.json',
[ INFO : 2025-12-31 15:10:08,116 ] -  'train_utt2spk': 'data/both/train-100/single.utt2spk',
[ INFO : 2025-12-31 15:10:08,116 ] -  'val_data': 'data/both/dev/raw.list',
[ INFO : 2025-12-31 15:10:08,116 ] -  'val_spk1_enroll': 'data/both/dev/spk1.enroll',
[ INFO : 2025-12-31 15:10:08,116 ] -  'val_spk2_enroll': 'data/both/dev/spk2.enroll',
[ INFO : 2025-12-31 15:10:08,116 ] -  'val_spk2utt': 'data/both/dev/single.wav.scp'}
```

I have got the problem in epoch 8. After 7 nice epochs, the loss turns nan. I cannot increase my batch size, maybe activate mixed precision. I will try that one, but do you have any other thought?
